src/source-napatech.c

```diff
@@ -149,81 +164,108 @@ void TmModuleNapatechDecodeRegister(void)
  * \param data      data pointer gets populated with
  *
  */
-TmEcode NapatechStreamThreadInit(ThreadVars *tv, const void *initdata, void **data)
-{
+TmEcode NapatechStreamThreadInit(ThreadVars *tv, const void *initdata, void **data) {
     SCEnter();
-    struct NapatechStreamDevConf *conf = (struct NapatechStreamDevConf *)initdata;
+    struct NapatechStreamDevConf *conf = (struct NapatechStreamDevConf *) initdata;
     uintmax_t stream_id = conf->stream_id;
     *data = NULL;
+    
+    if (tv->id == 1) {
+        SC_ATOMIC_INIT(total_packets); 
```

Comment:
I think it's better to move this to a place where threads aren't active yet, such as TmModuleNapatechStreamRegister